jedi/inference/compiled/access.py

```diff
@@ -504,7 +504,8 @@ def _get_signature(self):
 
     def get_return_annotation(self):
         try:
-            o = self._obj.__annotations__.get('return')
+            import typing
+            o = typing.get_type_hints(self._obj).get('return')
```

Comment:
Good idea. I didn't really know about this.

Reply:
Is there some way to get the original object ( let's call it `owner`) that the function that is being looked at( `_obj`) came from? To handle generics this is relevant, because otherwise this will just be a type variable like `{'return': ~T}` and `owner.__parameters__` and `owner.__orig_class__.__args__` is required to resolve this. There is no nice API call for that yet and `__orig_class__` is technically an "undocumented internal attribute", so if this is too much work, I don't think it is worth the hassle for now.

Reply:
Not that I know of. For classes in general it's pretty hard to figure out their context unfortunately.

For functions it's a bit easier. For functions you can use stuff like `func.__globals__` or in case of a method `func.__method__.__globals__`. There's also the `__code__` object on every function, where you have stuff like `co_filename`, `co_firstlineno`. For classes there's unfortunately mostly nothing, it's just `Class.__module__`.

Reply:
I can't think of a situation where the context of a class (nor class instance) would be needed to resolve generics.

There is `_obj.__self__` for bound methods which should easily cover the [case](https://mypy.readthedocs.io/en/latest/generics.html#defining-generic-classes)  I was thinking of. For generic functions the input arguments would need to be checked, but that definitely sounds like more work. And not something I really needed so far.